src/ui/SmartSnippet/SmartSnippet.ts

```diff
@@ -88,31 +122,31 @@ export class SmartSnippet extends Component {
   /**
    * @warning This method only works for the demo. In practice, the source of the answer will not always be part of the results.
    */
-  private getCorrespondingResult(questionAnswer: IQueryAnswerResponse) {
+  private getCorrespondingResult(questionAnswer: IQuestionAnswerResponse) {
     return find(
-      this.queryController.getLastResults().results,
+      this.lastResults.results,
       result => result.raw[questionAnswer.documentId.contentIdKey] === questionAnswer.documentId.contentIdValue
     );
   }
 
   private handleQuerySuccess(data: IQuerySuccessEventArgs) {
-    const { questionAnswer } = data.results;
+    const { questionAnswer } = (this.lastResults = data.results);
```

Comment:
You can retrieve the `lastResults` with `this.queryController.getLastResults()` and remove the property in the class.

Reply:
Good point! I'll replace it.

I was hesitant at first because I was worried that another query could be retrieved. However, all of this is executed right after the query success, so that seems impossible.